services/ui-src/src/components/fields/DynamicField.tsx

```diff
@@ -0,0 +1,88 @@
+import { useEffect } from "react";
+import { useFieldArray, useFormContext } from "react-hook-form";
+// components
+import { Box, Button, Flex, Image } from "@chakra-ui/react";
+// utils
+import { makeMediaQueryClasses } from "utils";
+// assets
+import cancelIcon from "assets/icons/icon_cancel_x_circle.png";
+import { TextField } from "./TextField";
+
+export const DynamicField = ({ name, label }: Props) => {
+  const mqClasses = makeMediaQueryClasses();
+
+  const form = useFormContext();
+  form.register(name);
+
+  const { fields, append, remove } = useFieldArray({
+    name: name,
+    shouldUnregister: true,
+  });
+
+  useEffect(() => {
+    if (fields.length === 0) append("");
+  });
+
+  return (
+    <Box sx={sx} className={mqClasses}>
+      <Box>
+        {fields.map((field: any, index: any) => {
+          return (
+            <Flex key={field.id} alignItems="flex-end">
+              <TextField
+                name={`${name}[${index}]`}
+                label={label}
+                dynamic={{
+                  parentName: name,
+                  index,
+                  inputRef: () => form.register,
+                }}
+              />
+              {index != 0 && (
+                <Button onClick={() => remove(index)} variant="unstyled">
+                  <Image
+                    sx={sx.removeButton}
+                    src={cancelIcon}
+                    alt="Remove item"
+                  />
+                </Button>
+              )}
```

Comment:
There was a bit of styling weirdness with this button where the button had a big wrapper around the icon that made the cursor into a pointer too soon. I started writing out a big thing explaining how to fix it but it was a pain so I just made a PR. I'll put that up soon. 

The first part is not making this a Chakra button and just using a regular HTML one.

Also I'm torn on `index != 0 && ...` -- technically it could just be `index && ...`. But I think it's better this way. Maybe add a comment that explains it's because we never want to have none and that's why it's not on the first one?

```suggestion
              {index != 0 && (
                <button onClick={() => remove(index)}>
                  <Image
                    sx={sx.removeButton}
                    src={cancelIcon}
                    alt="Remove item"
                  />
                </button>
              )}
```

Reply:
hmm I tried just `index &&` and it printed the index 0 next to the box. Could be something I missed. I just left it as is. Will add a comment. updated the button.